src/probability_theory/density.lean

```diff
@@ -0,0 +1,322 @@
+  quasi_measure_preserving X ℙ μ :=
+{ measurable := has_pdf.measurable X ℙ μ,
+  absolutely_continuous :=
+  begin
+    rw map_eq_with_density_pdf X ℙ μ,
+    exact with_density_absolutely_continuous _ _,
+  end }
+
+lemma map_absolutely_continuous {X : α → E} [has_pdf X ℙ μ] :
+  map X ℙ ≪ μ :=
+to_quasi_measure_preserving.absolutely_continuous
+
+lemma have_lebesgue_decomposition_of_has_pdf {X : α → E} [hX' : has_pdf X ℙ μ] :
+  (map X ℙ).have_lebesgue_decomposition μ :=
+⟨⟨⟨0, pdf X ℙ μ⟩,
+  by simp only [zero_add, measurable_pdf X ℙ μ, true_and, mutually_singular.zero.symm,
+    map_eq_with_density_pdf X ℙ μ] ⟩⟩
+
+lemma has_pdf_iff {X : α → E} (hX : measurable X) :
+  has_pdf X ℙ μ ↔ (map X ℙ).have_lebesgue_decomposition μ ∧ map X ℙ ≪ μ :=
+begin
+  split,
+  { intro hX',
+    exactI ⟨have_lebesgue_decomposition_of_has_pdf, map_absolutely_continuous⟩, },
+  { rintros ⟨h_decomp, h⟩,
+    haveI := h_decomp,
+    refine ⟨⟨hX, (measure.map X ℙ).rn_deriv μ, measurable_rn_deriv _ _, _⟩⟩,
+    rwa with_density_rn_deriv_eq }
+end
+
+section
+
+variables {F : Type*} [normed_group F] [measurable_space F] [second_countable_topology F]
+  [normed_space ℝ F] [complete_space F] [borel_space F] {ν : measure F}
+
+/-- A random variable that `has_pdf` transformed under a `quasi_measure_preserving`
+map also `has_pdf` if `(map g (map X ℙ)).have_lebesgue_decomposition μ`.
+
+`quasi_measure_preserving_has_pdf'` is more useful in the case we are working with a
+probability measure and a real-valued random variable. -/
+lemma quasi_measure_preserving_has_pdf {X : α → E} [has_pdf X ℙ μ]
+  {g : E → F} (hg : quasi_measure_preserving g μ ν)
+  (hmap : (map g (map X ℙ)).have_lebesgue_decomposition ν) :
+  has_pdf (g ∘ X) ℙ ν :=
+begin
+  rw [has_pdf_iff (hg.measurable.comp (has_pdf.measurable X ℙ μ)),
+      ← map_map hg.measurable (has_pdf.measurable X ℙ μ)],
+  refine ⟨hmap, _⟩,
+  rw [map_eq_with_density_pdf X ℙ μ],
+  refine absolutely_continuous.mk (λ s hsm hs, _),
+  rw [map_apply hg.measurable hsm, with_density_apply _ (hg.measurable hsm)],
+  have := hg.absolutely_continuous hs,
+  rw map_apply hg.measurable hsm at this,
+  exact set_lintegral_measure_zero _ _ this,
+end
+
+lemma quasi_measure_preserving_has_pdf' [is_finite_measure ℙ] [sigma_finite ν]
+  {X : α → E} [has_pdf X ℙ μ] {g : E → F} (hg : quasi_measure_preserving g μ ν) :
+  has_pdf (g ∘ X) ℙ ν :=
+begin
+  haveI : is_finite_measure (map g (map X ℙ)) :=
+    @is_finite_measure_map _ _ _ _ (map X ℙ)
+      (is_finite_measure_map ℙ (has_pdf.measurable X ℙ μ)) _ hg.measurable,
+  exact quasi_measure_preserving_has_pdf hg infer_instance,
+end
+
+end
+
+section real
+
+variables [is_finite_measure ℙ] {X : α → ℝ}
+
+/-- A real-valued random variable `X` `has_pdf X ℙ λ` (where `λ` is the Lebesgue measure) if and
+only if the push-forward measure of `ℙ` along `X` is absolutely continuous with respect to `λ`. -/
+lemma real.has_pdf_iff (hX : measurable X) : has_pdf X ℙ ↔ map X ℙ ≪ volume :=
```

Comment:
the measurable hypothesis could also be put in the right hand side of the equivalence. But the lemma you have here could be more useful if we work only with measurable functions. You could have both?